lsr_role2collection.py

```diff
@@ -868,11 +879,11 @@ def add_rolename(filename, rolename):
         if extra.name == "roles":
             for sr in extra.iterdir():
                 # If a role name contains '.', replace it with replace_dot
-                # convert nested subroles to prefix name with __ if not.
-                if sr.name.startswith("__"):
-                    dr = sr.name.replace(".", replace_dot)
-                else:
-                    dr = "__" + sr.name.replace(".", replace_dot)
+                # convert nested subroles to prefix name with subrole_prefix.
+                if sr.name.startswith("_"):
+                    sr.name = re.sub("^_*", "", sr.name)
```

Comment:
Are you trying to handle the case where a role may have a sub-role name that begins with `_`?  If so, we should assume the user knows what they are doing - do not transform it to have the `subrole_prefix`, and add it to the `no_readme_link` list - that is, assume the user is trying to denote the sub-role as a non-public sub-role.

Reply:
Could there be a case some sub-roles start with `_` and others do with `__`, and we want to make them consistent either starting with `_` or `__`?

Public sub-role vs. non-public sub-role, could there be a good indication? Currently, just metrics and logging have sub-role. Metrics is moving pcp to a separate collection. So, I thought it's better to be a non-public since it's going to go away. For rsyslog, we have no plan to make it public.

Reply:
I think it will be quite rare for role developers to use sub-roles, and the system roles team should provide guidance/pr reviews/etc. about naming.  If someone really, really wants to create a sub-role name that begins with `_` or `__` for the purpose of denoting that it is private, and they are not persuaded by our arguments to the contrary, then we should allow it, and not attempt to rename it when we convert the role to a collection.

Reply:
Ok. Then, is it ok to add subrole_prefix (`_` or `__`) only when the sub-role name does not start with `_`?

Reply:
> Ok. Then, is it ok to add subrole_prefix (`_` or `__`) only when the sub-role name does not start with `_`?

Yes.